src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -0,0 +1,192 @@
+package org.broadinstitute.dsde.rawls.jobexec
+
+import java.util.UUID
+
+import akka.actor._
+import akka.pattern._
+import com.google.api.client.auth.oauth2.Credential
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.RawlsException
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, SlickWorkspaceContext, ExecutionServiceDAO, SlickDataSource}
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{WorkspaceRecord, DataAccess, WorkflowRecord}
+import org.broadinstitute.dsde.rawls.jobexec.WorkflowSubmissionActor._
+import org.broadinstitute.dsde.rawls.model._
+import org.broadinstitute.dsde.rawls.util.FutureSupport
+import slick.dbio.DBIOAction
+import spray.http.OAuth2BearerToken
+
+import scala.concurrent.{Future, ExecutionContext}
+import scala.concurrent.duration.FiniteDuration
+
+
+object WorkflowSubmissionActor {
+  def props(datasource: SlickDataSource,
+            googleServicesDAO: GoogleServicesDAO,
+            executionServiceDAO: ExecutionServiceDAO,
+            batchSize: Int,
+            credential: Credential,
+            pollInterval: FiniteDuration): Props = {
+    Props(new WorkflowSubmissionActor(datasource, googleServicesDAO, executionServiceDAO, batchSize, credential, pollInterval))
+  }
+
+  sealed trait WorkflowSubmissionMessage
+  case object LookForWorkflows extends WorkflowSubmissionMessage
+  case class SubmitWorkflowBatch(workflowIds: Seq[Long]) extends WorkflowSubmissionMessage
+
+}
+
+class WorkflowSubmissionActor(val datasource: SlickDataSource,
+                              val googleServicesDAO: GoogleServicesDAO,
+                              val executionServiceDAO: ExecutionServiceDAO,
+                              val batchSize: Int,
+                              val credential: Credential,
+                              val pollInterval: FiniteDuration) extends Actor with WorkflowSubmission with LazyLogging {
+
+  import context._
+
+  scheduleNextWorkflowQuery
+
+  override def receive = {
```

Comment:
all the functions called in the receive should return Futures. The Futures must be piped back to the actor. More comments below